subprojects/parseq/src/test/java/com/linkedin/parseq/AbstractTaskTest.java

```diff
@@ -96,6 +97,34 @@ public void testRecover(int expectedNumberOfTasks) {
     runAndWait("AbstractTaskTest.testRecoverFailure", failure);
     assertEquals((int) failure.get(), -1);
     assertEquals(countTasks(failure.getTrace()), expectedNumberOfTasks);
+
+    Task<Integer> recoveredWithExceptionFilter = getFailureTask()
```

Comment:
To keep this test case from becoming too bloated, you should split this up into separate test methods.

Reply:
Test cases for 'recovering with exception filter' were extracted into separate `@Test` method.